locust_plugins/listeners.py

```diff
@@ -183,11 +183,11 @@ def _log_request(self, request_type, name, response_time, response_length, succe
 
         self._samples.append(sample)
 
-    def request_success(self, request_type, name, response_time, response_length, **_kwargs):
-        self._log_request(request_type, name, response_time, response_length, 1, None)
-
-    def request_failure(self, request_type, name, response_time, response_length, exception, **_kwargs):
-        self._log_request(request_type, name, response_time, response_length, 0, exception)
+    def on_request(self, request_type, name, response_time, response_length, exception, context, **_kwargs):
+        if exception:
+            self._log_request(request_type, name, response_time, response_length, 0, exception, context)
+        else:
+            self._log_request(request_type, name, response_time, response_length, 1, exception, context)
```

Comment:
No if statement needed. Actually we could just remove the success/fail flag (as it is the inverse of the presense of an exception). But for now we can just replace the 1/0 with a "exception is None")

Reply:
Also, you can remove (inline) the _log_request method, it was just used to merge failures/success:es into one